test/manual/mongodb_oidc.prose.test.ts

```diff
@@ -12,602 +27,1208 @@ describe('MONGODB-OIDC', function () {
-          // ``mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred``
-          // and the OIDC request callback.
+      describe('1.5  Multiple Principal No User', function () {
+        before(function () {
+          client = new MongoClient(
+            'mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred',
+            {
+              authMechanismProperties: {
+                REQUEST_TOKEN_CALLBACK: createRequestCallback()
+              }
+            }
+          );
+          collection = client.db('test').collection('test');
+        });
+
+        // Clear the cache.
+        // Create a client with a url of the form mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred and a valid OIDC request callback.
+        // Assert that a find operation fails.
+        // Close the client.
+        it('fails authentication', async function () {
+          try {
+            await collection.findOne();
+            expect.fail('Expected OIDC auth to fail with no user provided');
+          } catch (e) {
+            expect(e).to.be.instanceOf(MongoServerError);
+            expect(e.message).to.include('Authentication failed');
+          }
+        });
+      });
+
+      describe('1.6 Allowed Hosts Blocked', function () {
+        before(function () {
+          cache.clear();
+        });
+
+        // Clear the cache.
+        // Create a client that uses the OIDC url and a request callback, and an
+        // ``ALLOWED_HOSTS`` that is an empty list.
+        // Assert that a ``find`` operation fails with a client-side error.
+        // Close the client.
+        context('when ALLOWED_HOSTS is empty', function () {
           before(function () {
-            client = new MongoClient(
-              'mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred',
-              {
-                authMechanismProperties: {
-                  REQUEST_TOKEN_CALLBACK: requestCallback
-                }
+            client = new MongoClient('mongodb://localhost/?authMechanism=MONGODB-OIDC', {
+              authMechanismProperties: {
+                ALLOWED_HOSTS: [],
+                REQUEST_TOKEN_CALLBACK: createRequestCallback('test_user1', 600)
               }
-            );
+            });
             collection = client.db('test').collection('test');
           });
 
-          after(async () => {
-            client?.close();
-          });
-
-          // - Perform a ``find`` operation.
-          // - Clear the cache.
-          it('successfully authenticates', async function () {
-            const doc = await collection.findOne();
-            expect(doc).to.equal(null);
+          it('fails validation', async function () {
+            try {
+              await collection.findOne();
```

Comment:
This test won't fail if findOne returns successfully, see my other comment about adding a catch handler